generator/karaf/src/main/java/io/fabric8/maven/generator/karaf/KarafGenerator.java

```diff
@@ -71,6 +82,37 @@ public boolean isApplicable(List<ImageConfiguration> configs) {
                MavenUtil.hasPluginOfAnyGroupId(getProject(), KARAF_MAVEN_PLUGIN_ARTIFACT_ID);
     }
 
+    @Override
+    public BinaryInputArchiveBuilder getBinaryInputArchiveBuilder(final BuildService.BuildServiceConfig config) {
+        return new BinaryInputArchiveBuilder() {
+            @Override
+            public void createBinaryInput(ServiceHub hub, File binaryInputDirectory, ImageConfiguration imageConfiguration) throws Fabric8ServiceException {
+                try {
+                    File targetDirectory = new File(config.getBuildDirectory());
```

Comment:
The code here and in the WebApp generator look quite the same. Shouldn't we extract it into an external class which then is used here and in the Webapp generator ?